From 309e174cb4ca6510d3677c1b37ff6681b9fe5933 Mon Sep 17 00:00:00 2001
From: Romaric Jodin <rjodin@google.com>
Date: Tue, 27 Oct 2026 10:30:12 +0100
Subject: [PATCH] panfrost/kraid: Pass explicit XML file path in
 kraid_isa_xml_path

Pass the full path to isa-v9-v14.xml via --cfg kraid_isa_xml_path in
meson.build instead of only the directory path so the build system knows
which XML file is consumed.

Change-Id: I4fd2851c9fd5d5f4f6ed42546e9938ddab5cca36
---
 src/panfrost/compiler/kraid/isa.rs         |  4 ++--
 src/panfrost/compiler/kraid/meson.build    |  2 +-
 src/panfrost/compiler/kraid/proc/macros.rs | 20 +++++---------------
 3 files changed, 8 insertions(+), 18 deletions(-)

diff --git a/src/panfrost/compiler/kraid/isa.rs b/src/panfrost/compiler/kraid/isa.rs
index d69f54dce96e..f587359575b6 100644
--- a/src/panfrost/compiler/kraid/isa.rs
+++ b/src/panfrost/compiler/kraid/isa.rs
@@ -267,7 +267,7 @@ pub struct SrWriteLanes<L: Copy> {
 
 pub mod v9 {
     use kraid_proc_macros::*;
-    gen_isa_encode!("isa-v9-v14.xml", 9..=14);
+    gen_isa_encode!(9..=14);
 
     #[cfg(kraid_disasm)]
     pub mod decode {
@@ -419,6 +419,6 @@ pub mod v9 {
         type SourceEncoding = SourceEncodingX<false>;
         type SourceEncoding64 = SourceEncodingX<true>;
 
-        gen_isa_decode!("isa-v9-v14.xml", 9..=14);
+        gen_isa_decode!(9..=14);
     }
 }
diff --git a/src/panfrost/compiler/kraid/meson.build b/src/panfrost/compiler/kraid/meson.build
index 7e267199b60b..dd032fbfc60c 100644
--- a/src/panfrost/compiler/kraid/meson.build
+++ b/src/panfrost/compiler/kraid/meson.build
@@ -110,7 +110,7 @@ _libkraid_proc_macros_rs = rust.proc_macro(
 
 _libkraid_rust_args = [
   panfrost_rust_args,
-  '--cfg', 'kraid_isa_xml_path="' + meson.current_source_dir() + '"',
+  '--cfg', 'kraid_isa_xml_path="' + meson.current_source_dir() / 'isa-v9-v14.xml"',
   # Otherwise, rustc trips up on -pthread
   '-Clink-arg=-Wno-unused-command-line-argument',
 ]
diff --git a/src/panfrost/compiler/kraid/proc/macros.rs b/src/panfrost/compiler/kraid/proc/macros.rs
index d5c62724b3a2..0dc1384686c8 100644
--- a/src/panfrost/compiler/kraid/proc/macros.rs
+++ b/src/panfrost/compiler/kraid/proc/macros.rs
@@ -7,7 +7,6 @@ use compiler_proc::as_slice::*;
 use kraid_proc::*;
 use proc_macro::TokenStream;
 use std::ops::{Add, Range};
-use std::path::PathBuf;
 use std::str::FromStr;
 use syn::parse::{Parse, ParseStream};
 use syn::parse_macro_input;
@@ -87,18 +86,14 @@ impl Parse for LitRange {
 }
 
 struct GenIsaArgs {
-    xml: String,
     arch: Range<u8>,
 }
 
 impl Parse for GenIsaArgs {
     fn parse(input: ParseStream) -> syn::Result<Self> {
-        let xml: syn::LitStr = input.parse()?;
-        let _: syn::token::Comma = input.parse()?;
         let arch: LitRange = input.parse()?;
 
         Ok(GenIsaArgs {
-            xml: xml.value(),
             arch: arch.as_range(),
         })
     }
@@ -138,22 +133,17 @@ fn cfg_args() -> CfgArgs {
     CfgArgs { args }
 }
 
-fn cfg_isa_xml_path(args: &GenIsaArgs) -> String {
-    let isa_xml_path = cfg_args()
+fn cfg_isa_xml_path() -> String {
+    cfg_args()
         .find(|(key, _value)| key == "kraid_isa_xml_path")
         .and_then(|(_key, value)| value)
-        .expect("kraid_isa_xml_path not specified");
-
-    let mut xml_path = PathBuf::from(isa_xml_path);
-    xml_path.push(&args.xml);
-
-    String::from(xml_path.to_str().unwrap())
+        .expect("kraid_isa_xml_path not specified")
 }
 
 #[proc_macro]
 pub fn gen_isa_encode(item: TokenStream) -> TokenStream {
     let args = parse_macro_input!(item as GenIsaArgs);
-    let xml_path = cfg_isa_xml_path(&args);
+    let xml_path = cfg_isa_xml_path();
 
     isa::encoder::gen_encoder(&xml_path, args.arch)
         .unwrap()
@@ -163,7 +153,7 @@ pub fn gen_isa_encode(item: TokenStream) -> TokenStream {
 #[proc_macro]
 pub fn gen_isa_decode(item: TokenStream) -> TokenStream {
     let args = parse_macro_input!(item as GenIsaArgs);
-    let xml_path = cfg_isa_xml_path(&args);
+    let xml_path = cfg_isa_xml_path();
 
     isa::decoder::gen_decoder(&xml_path, args.arch)
         .unwrap()
-- 
GitLab
